src/Php81/Resources/stubs/CURLStringFile.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+if (\PHP_VERSION_ID >= 70400 && extension_loaded('curl') && class_exists(CURLFile::class, false)) {
+    /**
+     * @property string $data
+     */
+    class CURLStringFile extends CURLFile
+    {
+        private $data;
+
+        public function __construct(string $data, string $postname, string $mime = 'application/octet-stream')
+        {
+            $this->data = $data;
+            parent::__construct('data://application/octet-stream;base64,' . base64_encode($data), $mime, $postname);
+        }
+
+        public function __set(string $name, $value): void
+        {
```

Comment:
CURLStringFile has typed properties in PHP. So you probably need to validate the type of `$value` and trigger a TypeError when it is not a string